src/shipchain_common/rpc.py

```diff
@@ -45,25 +45,38 @@ def call(self, method, args=None):
 
         try:
             with TimingMetric('engine_rpc.call', tags={'method': method}) as timer:
-                response_json = settings.REQUESTS_SESSION.post(self.url,
-                                                               data=json.dumps(self.payload, cls=DecimalEncoder),
-                                                               timeout=getattr(settings, 'REQUESTS_TIMEOUT', 270)
-                                                               ).json()
+                response = settings.REQUESTS_SESSION.post(self.url, data=json.dumps(self.payload, cls=DecimalEncoder),
+                                                          timeout=getattr(settings, 'REQUESTS_TIMEOUT', 270))
+                response_json = response.json()
+
                 LOG.info('rpc_client(%s) duration: %.3f', method, timer.elapsed)
 
-            if 'error' in response_json:
+            if 'error' in response_json and response.status_code == status.HTTP_200_OK:
+                # It's an error properly handled by engine
                 log_metric('engine_rpc.error', tags={'method': method, 'code': response_json['error']['code'],
                                                      'module': __name__})
                 LOG.error('rpc_client(%s) error: %s', method, response_json['error'])
                 raise RPCError(response_json['error']['message'])
 
+            if status.HTTP_400_BAD_REQUEST <= response.status_code:
```

Comment:
This line is not very clear, and I think I'd prefer to use the DRF helpers under rest_framework.status (is_success, etc).